app/graphql/mutations/hide_submission.rb

```diff
@@ -0,0 +1,44 @@
+module Mutations
+  class HideSubmission < ApplicationQuery
+    argument :submission_id, String, required: true
+    argument :hide, Boolean, required: true
+
+    description "Hide or unhide a submission from discussion"
+
+    field :success, Boolean, null: false
+
+    def resolve(_params)
+      if @params[:hide]
+        submission.hidden_at = Time.zone.now
+        submission.hidden_by = current_user
+      else
+        submission.hidden_at = nil
+        submission.hidden_by = nil
+      end
+
+      submission.save!
+      { success: true }
+    end
+
+    def query_authorized?
+      return false if current_user.blank?
+
+      # school admin or course author
+      if current_school_admin.present? ||
+           current_user.course_authors.where(course: course).present?
```

Comment:
I doubt that course authors should be allowed to perform _moderation_ actions on discussion items - they're supposed to only have a _preview_ of the course material when they're interacting with the student side.

This comment applies to other moderation activities as well.

Reply:
I thought I heard in some discussion we mention the admins, authors and coaches needed the ability to do moderation actions 🤔 Maybe I miss heard. Will remove this.

Reply:
Removed course author from moderation actions.